### PR TITLE
Stylistic improvement

### PR DESCRIPTION
**Pull request by aki-nishimura**
Refactoring and minor changes to the code for stylistic improvement. None of them looks controversial to me but, again, I am requesting a quick review I don't know the historical decision processes.

### CODE REVIEW

File: R/harmonicHMC.R [hunk 1]
```diff
@@ -59,11 +61,11 @@ harmonicHMC <- function(n,
   if (sum(F %*% init + g > 0) < length(g)) {
     stop("Initial value x does not satisfy Fx + g >=0")
   }
-  samples <- matrix(nrow = n + burnin, ncol = ncol(F))
+  samples <- matrix(nrow = nSample, ncol = ncol(F))
   randomBounceTime <-
     ifelse(length(time) == 2, TRUE, FALSE)
   bounceDistances <- vector(mode = "list",
-                            length = ifelse(diagnosticMode, n + burnin, 0))
+                            length = ifelse(diagnosticMode, nSample + burnin, 0))
```
Comment (chinandrew) [inline R/harmonicHMC.R:68]: In this case, is it odd to return bounce distances for all samples, including burn in, but not return the burn in samples.

File: R/zigzagHMC.R [hunk 2]
```diff
@@ -11,36 +11,38 @@
...
 #' @param init a d-dimensional vector of the initial value. `init` must satisfy all constraints. If `init = NULL`, a random initial value will be used.
-#' @param step step size for Zigzag-HMC or Zigzag-NUTS (if `nutsFlg = TRUE`). Default value is the empirically optimal choice: sqrt(2)(lambda)^(-1/2) for Zigzag-HMC and 0.1(lambda)^(-1/2) for Zigzag-NUTS, where lambda is the minimal eigenvalue of the precision matrix.   
-#' @param rSeed random seed (default = 1).
+#' @param stepsize step size for Zigzag-HMC or Zigzag-NUTS (if `nutsFlg = TRUE`). Default value is the empirically optimal choice: sqrt(2)(lambda)^(-1/2) for Zigzag-HMC and 0.1(lambda)^(-1/2) for Zigzag-NUTS, where lambda is the minimal eigenvalue of the precision matrix.   
+#' @param seed random seed (default = 1).
+#' @param diagnosticMode logical. `TRUE` for also returning diagnostic information such as the stepsize used. 
 #'
-#' @return an (n + burnin)*d matrix of samples. The first `burnin` samples are from the user specified warm-up iterations.
+#' @return an nSample-by-d matrix of samples. If `diagnosticMode` is `TRUE`, a list with additional diagnostic information is returned.
```
Comment (chinandrew) [inline R/zigzagHMC.R:18]:
```suggestion
#' @return an nSample x d matrix of samples. If `diagnosticMode` is `TRUE`, a list with additional diagnostic information is returned. 
```

to be consistent with `harmonicHMC`'s  documentation. Or if this method is preferred, would want to change the sentence in `harmonicHMC`

File: R/zigzagHMC.R [hunk 3]
```diff
@@ -77,59 +78,59 @@ zigzagHMC <- function(n,
...
   } else {
-    if (!is.null(step)) {
-      t <- step
-    } else{
-      t <-
+    if (is.null(stepsize)) {
+      stepsize <-
         sqrt(2) / sqrt(min(mgcv::slanczos(
           A = prec, k = 1, kl = 1
         )[['values']], na.rm = T))
```
Comment (chinandrew) [inline R/zigzagHMC.R:105]: unrelated to this PR, but realized that one of these `slanczos` calls has `na.rm=T` while the other one doesn't, not sure if there was some reason for that.

File: R/harmonicHMC.R (hunk 1 above)
Comment (aki-nishimura) [reply R/harmonicHMC.R]: Good point, thanks. Fixed.

File: R/zigzagHMC.R (hunk 2 above)
Comment (aki-nishimura) [reply R/zigzagHMC.R:18]: I updated `harmonicHMC`'s doc b/c part of it already uses the "-by-" notation.  To be honest, though, the doc styles aren't consistent at all across `zigzagHMC` and `harmonicHMC`; I am ignoring it for now b/c they are just docs and are still readable.

File: R/zigzagHMC.R (hunk 3 above)
Comment (aki-nishimura) [reply R/zigzagHMC.R:105]: Interesting. Any reason @Zhenyu028?

Comment (Zhenyu028) [reply R/zigzagHMC.R:105]: good catch, I don't remember any convergence issue with `slanczos` and `slanczos(...)[['values']]` should be two numbers being the largest and smallest eigenvalues. Though this `na.rm = T` I wrote made me think there may be cases where the largest eigenvalue is null. We can add a check (that slanczos returns two real numbers) here?

Comment (aki-nishimura) [reply R/zigzagHMC.R:105]: OK, the convergence issue is less likely for the largest eigenvalue than for the smallest with Lanczos, but I'll watch out for potentially problematic behaviors.

**Pull request merged**